File: 796-rotate-string.py
# look at js file for reasoning

class Solution(object):
    def rotateString(self, s, goal):
        """
        :type s: str
        :type goal: str
        :rtype: bool
        """
        sLst = list(s)

        for char in sLst:
            shift = sLst[0]
            del sLst[0]
            sLst.append(shift)
            if "".join(sLst) == goal:
                return True

        return False

File: 796-rotate-string.js
// return true if s can become goal after a number of shifts on s
// a shift is moving leftmost character to rightmost

/*
0.) convert s to an array to iterate
1.) iterate for every character in s
2.) shift and push on each iteration, check if s array converted to string is goal
3.) return true if it is, false if the loop exits
*/

var rotateString = function(s, goal) {
    let sArr = s.split("");

    for (let i = 0; i < sArr.length; i++) {
        let shift = sArr.shift();
        sArr.push(shift);
        if (sArr.join("") === goal) return true;
    }

    return false;
};
